.github/workflows/build-publish.yaml

```diff
@@ -0,0 +1,141 @@
+name: Build & Publish
+
+on:
+  push:
+    branches: ["master"]
+    paths-ignore:
+      - "chart/Chart.yaml"
+      - "CHANGELOG.md"
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  test-chart:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Test chart
+        uses: silphid/actions/test-chart@v1
+        with:
+          work-dir: chart
+  test-app:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Go environment
+        uses: actions/setup-go@v3.3.1
+        with:
+          go-version-file: go.mod
+          cache: true
+          cache-dependency-path: go.sum
+      - name: Setup dependencies
+        run: go mod download
+      - name: Run tests
+        run: go test ./...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - uses: actions/checkout@v3
+      - name: Build Docker image
+        uses: docker/build-push-action@v3.2.0
+        with:
+          context: .
+          push: false
```

Comment:
Publish has a bunch of additional steps that don't need to be run during a PR build, so separating them makes for a cleaner build pipeline, IMO. And since we are using the GitHub actions layer cache, there is no major sacrifice in performance.